pkg/iris/workloads/services/mapper/cronjob.go

```diff
@@ -49,20 +51,24 @@ func (m *mapworkload) CreateCronjobEcst(clusterName string, cronJob batchv1.Cron
 
 func ResolveK8sServiceForK8sCronJob(services *v1.ServiceList, cronJob batchv1.CronJob) string {
 	cronJobService := ""
-	for _, service := range services.Items {
-		sharedLabelsCronJob := map[string]string{}
-		sharedLabelsService := map[string]string{}
-		for label := range service.Spec.Selector {
-			if _, ok := cronJob.Spec.JobTemplate.Spec.Selector.MatchLabels[label]; ok {
-				sharedLabelsCronJob[label] = cronJob.Spec.JobTemplate.Spec.Selector.MatchLabels[label]
-				sharedLabelsService[label] = service.Spec.Selector[label]
-			}
-		}
+	if cronJob.Spec.JobTemplate.Spec.Selector.MatchLabels != nil {
```

Comment:
you have to check if cronJob.Spec.JobTemplate.Spec.Selector is not nil because it's a pointer type it can be nil. MatchLabels is a struct so it won't be nil.

Reply:
@DennisBerthold-leanix , I still have a question here. Wouldn't be there a case where selector is not nil but matchLabels are nil ?

Reply:
Thanks! Now we are checking if selector is nil.